src/WireMock.Net/Server/WireMockServer.Admin.cs

```diff
@@ -600,6 +601,26 @@ private IResponseMessage RequestsFind(IRequestMessage requestMessage)
 
         return ToJson(result);
     }
+
+    private IResponseMessage RequestFindByMappingGuid(IRequestMessage requestMessage)
+    {
+        if (requestMessage.Query != null &&
+            requestMessage.Query.TryGetValue("mappingGuid", out var value) &&
+            Guid.TryParse(value.ToString(), out var mappingGuid)
+        )
+        {
+            var logEntry = LogEntries.SingleOrDefault(le => !le.RequestMessage.Path.StartsWith("/__admin/") && le.MappingGuid == mappingGuid);
+            if (logEntry != null)
+            {
+                var logEntryMapper = new LogEntryMapper(_options);
+                return ToJson(logEntryMapper.Map(logEntry));
+            }
+
+            return ResponseMessageBuilder.Create(HttpStatusCode.OK);

```

Comment:
Returning 200 with an empty response is probably not what the client would expect. If the status code is 200, the client will expect the body to contain a `LogEntryModel`.

But anyway, I think this endpoint should return a collection (which will be empty if no match is found)